Mathlib/CategoryTheory/ConcreteCategory/Bundled.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2018 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison, Johannes Hölzl, Reid Barton, Sean Leather
+-/
+import Std.Tactic.Lint.Frontend
+import Std.Tactic.Lint.Misc
+
+/-!
+# Bundled types
+
+`bundled c` provides a uniform structure for bundling a type equipped with a type class.
+
+We provide `category` instances for these in `category_theory/unbundled_hom.lean`
+(for categories with unbundled homs, e.g. topological spaces)
+and in `category_theory/bundled_hom.lean` (for categories with bundled homs, e.g. monoids).
+-/
+
+universe u v
+
+namespace CategoryTheory
+
+variable {c d : Type u → Type v} {α : Type u}
+
+/-- `bundled` is a type bundled with a type class instance for that type. Only
+the type class is exposed as a parameter. -/
+structure Bundled (c : Type u → Type v) : Type max (u + 1) v where
+  /-- The underlying type of the bundled type -/
+  α : Type u
+  /-- The corresponding instance of the bundled type class -/
+  str : c α := by infer_instance
+
+namespace Bundled
+
+-- This is needed so that we can ask for an instance of `c α` below even though Lean doens't know
+-- that `c α` is a typeclass.
+set_option checkBinderAnnotations false
```

Comment:
```suggestion
set_option checkBinderAnnotations false in
```

Then you don't need the `set_option false` below.

Reply:
Thanks, I didn't know about that.